core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -174,14 +174,24 @@ class CostBasedRegionPlanGenerator(
     * The core of the search algorithm. If the input physical plan is already schedulable, no search will be executed.
     * Otherwise, depending on the configuration, either a global search or a greedy search will be performed to find
     * an optimal plan. The search starts from a plan where all non-blocking edges are pipelined, and leads to a low-cost
-    * schedulable plan. Optimizations based on chains and bridges are included in the search.
+    * schedulable plan by changing pipelined non-blocking edges to materialized. By default all pruning techniques
+    * are enabled (chains, clean edges, and early stopping on schedulable states).
```

Comment:
I only see two booleans for optimizations on argument. are we missing the `early stopping` one?

Reply:
That will be added in the next PR.

Reply:
then mention it in the next PR?

Reply:
Yes, I already removed this in the current documentation.